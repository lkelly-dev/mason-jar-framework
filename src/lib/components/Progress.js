import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';
import { debugAttrs, debugMode } from '../utils/debugging';

const Progress = props => {
  const { context, children, className, currentValue, min, max, size, ...rest } = props;
  const debug = debugMode(context)
  const progressSize = MODIFIERS.SIZE[size] ? `progress-${size}` : null;
  const progressClass = cx('progress', progressSize, className);
  return (
    <div className={progressClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      <div
        style={{ width: `${currentValue}%` }}
        className="progress-bar"
        role="progressbar"
        aria-valuenow={currentValue}
        aria-valuemin={min || 0}
        aria-valuemax={max || 100}
      />
    </div>
  );
};

export { Progress };

Progress.propTypes = {
  currentValue: PropTypes.number.isRequired,
};

//  min: PropTypes.number,
// max: PropTypes.number,
