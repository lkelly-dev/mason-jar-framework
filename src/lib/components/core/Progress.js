import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';

const Progress = props => {
  const { children, className, now, min, max, size, ...rest } = props;
  const progressSize = MODIFIERS.SIZE[size] ? `progress-${size}` : null;
  const progressClass = cx('progress', progressSize, className);
  return (
    <div className={progressClass} {...rest}>
      <div
        style={{ width: `${now}%` }}
        className="progress-bar"
        role="progressbar"
        aria-valuenow={now}
        aria-valuemin={min || 0}
        aria-valuemax={max || 100}
      />
    </div>
  );
};

export { Progress };

Progress.propTypes = {
  now: PropTypes.number.isRequired,
};

//  min: PropTypes.number,
// max: PropTypes.number,
