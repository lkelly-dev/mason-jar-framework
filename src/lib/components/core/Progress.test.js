import React from 'react';
import { shallow } from 'enzyme';
import { Progress } from "./Progress";

describe('Progress', () => {
  it('Progress matches snapshot', () => {
    const component = shallow(<Progress now={ 0 }></Progress>);
    expect(component).toMatchSnapshot();
  });
  it('has expected props', () => {
    // FIXME this component may need refactoring
    const cn = 'neato',
          size = 'sm',
          now = 0,
          min = 0,
          max = 100;

    const component = shallow(
      <Progress
        what="ever"
        className={ cn }
        size={ size }
        now={ now }
        min={ min }
        max={ max }>
      </Progress>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toContain(`progress-${size}`);
    expect(component.prop('what')).toBe('ever');
    const child = component.find('div.progress-bar');
    expect(child.prop('className')).toBe('progress-bar');
    expect(child.prop('role')).toBe('progressbar');
    expect(child.prop('aria-valuenow')).toBe(0);
    expect(child.prop('aria-valuemin')).toBe(0);
    expect(child.prop('aria-valuemax')).toBe(100);
  });
});
