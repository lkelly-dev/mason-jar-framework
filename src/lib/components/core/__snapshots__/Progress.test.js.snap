// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Progress matches snapshot 1`] = `
<div
  className="progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;
